File: Cs/Solution/CountPoints.cs
namespace Solution
{
	public partial class Solution
	{
		public int[] CountPoints(int[][] points, int[][] queries)
		{
			int[] ret = new int[queries.Length];
			for (int i = 0; i < queries.Length; i++)
			{
				int[] query = queries[i];
				int x = query[0];
				int y = query[1];
				int r = query[2];
				int count = 0;
				foreach (int[] point in points)
				{
					int px = point[0];
					int py = point[1];
					if ((px - x) * (px - x) + (py - y) * (py - y) <= r * r)
					{
						count++;
					}

				}
				ret[i] = count;
			}
			return ret;
		}
	}
}

File: Ruby/solution/count_points.rb
# @param {Integer[][]} points
# @param {Integer[][]} queries
# @return {Integer[]}
def count_points(points, queries)
	queries.map{|q| points.count{|p| ((p[0]-q[0]) * (p[0]-q[0]) + (p[1]-q[1]) * (p[1]-q[01])) <= (q[2] * q[2])}}
end

input0 = [[[1,3],[3,3],[5,3],[2,2]], [[1,1],[2,2],[3,3],[4,4],[5,5]]]
input1 = [[[2,3,1],[4,3,1],[1,1,2]], [[1,2,2],[2,2,2],[4,3,2],[4,3,3]]]
expect = [[3,2,2], [2,3,2,4]]

fail_cnt = 0
pass_cnt = 0

puts "\n"
for i in 0...input0.length
        output = count_points(input0[i], input1[i])
        if expect[i] != output
                fail_cnt += 1
                puts "Input #{i} expect #{expect[i]}, actual #{output}"
        else
                pass_cnt += 1
        end
end

puts "==== Test Summary ===="
if fail_cnt == 0
        puts "TEST PASS!"
else
        puts "TEST FAIL!"
end
puts "test #{fail_cnt+pass_cnt}"
puts "pass #{pass_cnt}"
puts "fail #{fail_cnt}"
puts "======================"
